### ISSUE TITLE
still cant build actual version from PHP-src

### ISSUE THREAD

**Issue by Kapsonfire-DE** [swoole/swoole-src#3580]
Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a simple script for reproducing the error.
Followed these steps https://github.com/Kapsonfire-DE/OpenTwitchAssistent/wiki/Using-PHP-on-Windows-10-with-linux-binaries-using-WSL except got PHP8 from PHP-src instead of beta1

2. What did you expect to see?
Built .so

3. What did you see instead?
`/home/kapsonfire/swoole/swoole-src/swoole_event.cc:322:85: error: ‘php_sockets_le_socket’ was not declared in this scope
             if ((php_sock = (php_socket *) zend_fetch_resource_ex(zsocket, nullptr, php_sockets_le_socket()))) {
                                                                                     ^~~~~~~~~~~~~~~~~~~~~
/home/kapsonfire/swoole/swoole-src/swoole_event.cc:322:85: note: suggested alternative: ‘php_swoole_get_socket’
             if ((php_sock = (php_socket *) zend_fetch_resource_ex(zsocket, nullptr, php_sockets_le_socket()))) {
                                                                                     ^~~~~~~~~~~~~~~~~~~~~
                                                                                     php_swoole_get_socket
/home/kapsonfire/swoole/swoole-src/swoole_event.cc: In function ‘int swoole_convert_to_fd_ex(zval*, int*)’:
/home/kapsonfire/swoole/swoole-src/swoole_event.cc:376:85: error: ‘php_sockets_le_socket’ was not declared in this scope
             if ((php_sock = (php_socket *) zend_fetch_resource_ex(zsocket, nullptr, php_sockets_le_socket()))) {
                                                                                  ^~~~~~~~~~~~~~~~~~~~~
/home/kapsonfire/swoole/swoole-src/swoole_event.cc:376:85: note: suggested alternative: ‘php_swoole_get_socket’
             if ((php_sock = (php_socket *) zend_fetch_resource_ex(zsocket, nullptr, php_sockets_le_socket()))) {
                                                                                     ^~~~~~~~~~~~~~~~~~~~~
                                                                                     php_swoole_get_socket
/home/kapsonfire/swoole/swoole-src/swoole_event.cc: In function ‘void swoole_php_socket_free(zval*)’:
/home/kapsonfire/swoole/swoole-src/swoole_event.cc:419:72: error: ‘php_sockets_le_socket’ was not declared in this scope
     php_sock = (php_socket *) zend_fetch_resource_ex(zsocket, nullptr, php_sockets_le_socket());
                                                                        ^~~~~~~~~~~~~~~~~~~~~
/home/kapsonfire/swoole/swoole-src/swoole_event.cc:419:72: note: suggested alternative: ‘php_swoole_get_socket’
     php_sock = (php_socket *) zend_fetch_resource_ex(zsocket, nullptr, php_sockets_le_socket());
                                                                        ^~~~~~~~~~~~~~~~~~~~~
                                                                        php_swoole_get_socket
make: *** [Makefile:379: swoole_event.lo] Error 1`

4. What version of Swoole are you using (show your `php --ri swoole`)?
see above

5. What is your machine environment used (show your `uname -a` & `php -v` & `gcc -v`) ?
Linux DESKTOP-77QA95M 4.4.0-19041-Microsoft #1-Microsoft Fri Dec 06 14:06:00 PST 2019 x86_64 GNU/Linux

PHP 8.0.0-dev (cli) (built: Aug 17 2020 11:51:48) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.0-dev, Copyright (c) Zend Technologies

Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 8.3.0-6' --with-bugurl=file:///usr/share/doc/gcc-8/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-8 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 8.3.0 (Debian 8.3.0-6)

Related to: https://github.com/swoole/swoole-src/issues/3565

**Comment by twose** [comment]
Please disable socket option when configure

**Comment by Kapsonfire-DE** [comment]
Yep - its running.
Thanks

Don't I need socket support for websocket client and so on?

**Comment by twose** [comment]
it is just for converting swoole-socket to php-ext-socket

**Comment by Kapsonfire-DE** [comment]
pros and cons?

**Comment by eldadfux** [comment]
@twose, @matyhtf, I'm also getting an error trying to build on PHP8 alpine (php:8.0.0beta1-cli-alpine) image (` syntax error: unexpected "(" (expecting "fi")`):

```
checking if awk is broken... no
checking enable debug log... no
checking enable trace log... no
checking enable sockets support... yes
checking enable openssl support... no
checking enable http2.0 support... yes
checking swoole support... yes, shared
checking enable mysqlnd support... no
checking dir of openssl... no
checking dir of jemalloc... no
checking enable asan... no
checking whether to enable swoole coverage support... no
checking whether to enable Swoole developer build flags... no
checking if compiling with clang... no
./configure: line 5578: syntax error: unexpected "(" (expecting "fi")
ERROR: Service 'appwrite' failed to build: The command '/bin/sh -c wget -q https://github.com/phpredis/phpredis/archive/$PHP_REDIS_VERSION.tar.gz &&   tar -xf $PHP_REDIS_VERSION.tar.gz &&   cd phpredis-$PHP_REDIS_VERSION &&   phpize &&   ./configure &&   make && make install &&   cd .. &&   git clone https://github.com/swoole/swoole-src.git &&   cd swoole-src &&   git checkout $PHP_SWOOLE_VERSION &&   phpize &&   ./configure --enable-sockets --enable-http2 &&   make && make install &&   cd ..' returned a non-zero code: 2
Eldads-MacBook-Pro-2:appwrite eldadfux$ 
```

This is my Docker file: https://github.com/appwrite/appwrite/blob/php8/Dockerfile (as mentioned before, same Dockerfile works with PHP7.4 latest image)

**Comment by twose** [comment]
> pros and cons?

Historical feature, useless now

**Comment by twose** [comment]
Fixed via #3684
